src/Analyzers/CSharp/Analyzers/ConvertProgram/ConvertProgramAnalysis_TopLevelStatements.cs

```diff
@@ -0,0 +1,159 @@
+            return true;
+        }
+
+        private static bool TypeCanBeConverted(INamedTypeSymbol containingType, TypeDeclarationSyntax typeDeclaration)
+        {
+            // Can't convert if our Program type derives or implements anything special.
+            if (containingType.BaseType?.SpecialType != SpecialType.System_Object)
+                return false;
+
+            if (containingType.AllInterfaces.Length > 0)
+                return false;
+
+            // Too complex to convert many parts to top-level statements.  Just bail on this for now.
+            if (containingType.DeclaringSyntaxReferences.Length > 1)
+                return false;
+
+            // Too complex to support converting a nested type.
+            if (containingType.ContainingType != null)
+                return false;
+
+            // If the type wasn't internal it might have been public and something outside this assembly might be using it.
+            if (containingType.DeclaredAccessibility != Accessibility.Internal)
+                return false;
+
+            // type can't be converted with attributes.
+            if (typeDeclaration.AttributeLists.Count > 0)
+                return false;
+
+            // can't convert doc comments to top level statements.
+            if (typeDeclaration.GetLeadingTrivia().Any(t => t.IsDocComment()))
+                return false;
+
+            // All the members of the type need to be private/static.  And we can only have fields or methods. that's to
+            // ensure that no one else was calling into this type, and that we can convert everything in the type to
+            // either locals or local-functions.
+
+            foreach (var member in typeDeclaration.Members)
+            {
+                // method can't be converted with attributes.
+                if (member.AttributeLists.Count > 0)
+                    return false;
+
+                // if not private, can't convert as something may be referencing it.
+                if (member.Modifiers.Any(m => m.Kind() is SyntaxKind.PublicKeyword or SyntaxKind.ProtectedKeyword or SyntaxKind.InternalKeyword))
+                    return false;
+
+                if (!member.Modifiers.Any(SyntaxKind.StaticKeyword))
+                    return false;
+
+                if (member is not FieldDeclarationSyntax and not MethodDeclarationSyntax)
+                    return false;
+
+                if (member is MethodDeclarationSyntax methodDeclaration)
+                {
+                    // if a method, it has to actually have a body so we can convert it to a local function.
+                    if (methodDeclaration is { Body: null, ExpressionBody: null })
+                        return false;
+
+                    // local functions can't be unsafe
+                    if (methodDeclaration.Modifiers.Any(SyntaxKind.UnsafeKeyword))
+                        return false;
```

Comment:
They can be unsafe, I think

Reply:
oh, you're right. not sure why i thought that.  will fix.